src/main/java/com/demonwav/mcdev/platform/forge/inspections/sideonly/SideOnlyProjectComponent.java

```diff
@@ -53,25 +55,19 @@ public void projectOpened() {
                     public void run(@NotNull ProgressIndicator indicator) {
                         try (final AccessToken ignored = ApplicationManager.getApplication().acquireReadActionLock()) {
                             indicator.setIndeterminate(true);
-                            final JavaRecursiveElementWalkingVisitor visitor = new JavaRecursiveElementWalkingVisitor() {
-                                @Override
-                                public void visitField(PsiField field) {
-                                    super.visitField(field);
-                                    SidedProxyAnnotator.check(field);
-                                }
-                            };
-
-                            try {
-                                final AllClassesSearchExecutor executor = new AllClassesSearchExecutor();
-                                executor.execute(
-                                    new AllClassesSearch.SearchParameters(GlobalSearchScope.projectScope(myProject), myProject),
-                                    psiClass -> {
-                                        psiClass.acceptChildren(visitor);
-                                        return true;
-                                    }
-                                );
-                            } catch (Exception ignored2) {}// I have no idea, seems like it's a bug with intellij, but I'll catch any
-                                                           // weird exceptions that pop up anyways...
+                            GlobalSearchScope scope = GlobalSearchScope.projectScope(myProject);
+                            PsiClass sidedProxy = JavaPsiFacade.getInstance(myProject).findClass(ForgeConstants.SIDED_PROXY_ANNOTATION, scope);
+                            if (sidedProxy == null) {
+                                return;
+                            }
+                            Collection<PsiField> annotatedFields = AnnotatedElementsSearch.searchElements(sidedProxy, scope, PsiField.class).findAll();
```

Comment:
You can use `AnnotatedElementsSearch.searchPsiFields` here instead of adding `PsiField.class` explicitely